libraries/azure-app-configuration-importer/src/internal/parsers/defaultConfigurationSettingsConverter.ts

```diff
@@ -32,43 +34,68 @@ export class DefaultConfigurationSettingsConverter implements ConfigurationSetti
   public Convert(
     config: object,
     options: SourceOptions
-  ): SetConfigurationSettingParam<string | FeatureFlagValue | SecretReferenceValue>[] {
+  ): SetConfigurationSettingParam<string | SecretReferenceValue>[] {
     let configurationSettings = new Array<
-      SetConfigurationSettingParam<string | FeatureFlagValue>
+      SetConfigurationSettingParam
     >();
-    let featureFlagsConfigSettings = new Array<SetConfigurationSettingParam<FeatureFlagValue>>();
-    let foundFeatureManagement = false;
-    let featureFlags: any = {};
-    let featureManagementKeyWord = "";
-    let enabledForKeyWord = "";
+
+    let featureFlagsConfigSettings = new Array<SetConfigurationSettingParam>();
+    let foundMsFmSchema = false;
+    let foundLegacySchema = false;
+    let msFmFeatureManagementKeyWord = "";
+    let legacySchemaFeatureManagementKeyWord = "";
+    let legacySchemaEnabledForKeyWord = "";
+    let legachSchemaRequirementTypeKeyWord = "";
+
+    const featureFlagsDict: any = {};
 
     if (options.format == ConfigurationFormat.Properties && !options.skipFeatureFlags) {
       this.checkFeatureManagementExist(config);
     }
 
-    for (let i = 0; i < Constants.FeatureManagementKeyWords.length; i++) {
+    for (let i = 0; i < Constants.FeatureManagementKeyWords.length - 1; i++) {
```

Comment:
Why do we have `i < Constants.FeatureManagementKeyWords.length -1` instead of `i < Constants.FeatureManagementKeyWords.length` ?

Reply:
See [this ](https://github.com/Azure/AppConfiguration-JavaScriptImporter/pull/13#discussion_r1824117977)